### PR TITLE
Allow passing in options for acorn

### PR DESCRIPTION
**Pull request by mjangda**
`acorn` supports various options, such as setting the ecma version for your source code (https://github.com/ternjs/acorn#main-parser). Allow passing in those options when instantiating the XGettext object.

This is needed for the WordPress.com Desktop app, which currently uses some newer ecma features in untranspiled mode (via `'use strict'`) and we need to be able to parse through and pull out translation strings properly.

/cc @deBhal @aduth

### CODE REVIEW

**Comment by mjangda** [comment]
For the desktop-specific ecma6 use case we could also bump to acorn 3.0.0 which defaults to 6: https://github.com/ternjs/acorn/commit/bed00427b5b3b3be57abef555ef1c466fc493f5b

File: test/xgettext.js [hunk 1]
```diff
@@ -94,3 +94,26 @@ it( 'should match functions that are the last element of a recursive sequence (c
...
+
+describe( '#parserOptions', function() {
+	it( 'with ecma6, should fail when parserOptions not set', function() {
+		var source = 'const i = 0; _( "Hello World!" );',
+			parser = new XGettext(),
+			fn = function() {
+				parser.getMatches( source );
+			};
+
+			expect( fn ).to.throw(SyntaxError);
```
Comment (aduth) [inline test/xgettext.js]: Tabbing off by one here.

File: xgettext.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ var XGettext = module.exports = function( options ) {
 	this.options = _.extend( {}, XGettext.defaultOptions, options );
 	this.options.keywords = this._normalizeKeywords( this.options.keywords );
 	this.options.keywordFunctions = Object.keys( this.options.keywords );
+	this.options.parserOptions = _.extend( {}, options.parserOptions );
```
Comment (aduth) [inline xgettext.js]: Do we need to create a copy of `parseOptions` ? This should already be injected into `this.options` above on line 15.

**Comment by aduth** [comment]
>we could also bump to acorn 3.0.0

Sounds like a better approach, though I'm uncertain what backwards-incompatible changes that might include.

File: xgettext.js (hunk 2 above)
Comment (deBhal) [inline xgettext.js]: Probably belongs in a separate PR, but if a user passes in `{locations:false}` we crash trying to pull properties out of `loc`.  We should probably rename it to something like 'requiredOptions', and add it in as the last argument to `_.extend()` on line 15 so it overwrites the user options.

**Comment by aduth** [comment]
Superceded by #9.
